dask/compose.py

```diff
@@ -0,0 +1,281 @@
+
+    Examples
+    --------
+    Can be used as a decorator:
+
+    >>> @do
+    ... def add(a, b):
+    ...     return a + b
+    >>> res = add(1, 2)
+    >>> type(res) == Value
+    True
+    >>> res.compute()
+    3
+
+    For other cases, it may be cleaner to call ``do`` on a function at call
+    time:
+
+    >>> res2 = do(sum)([res, 2, 3])
+    >>> res2.compute()
+    8
+    """
+    @wraps(func)
+    def _dfunc(*args, **kwargs):
+        return applyfunc(func, *args, **kwargs)
+    return _dfunc
+
+
+def optimize(dsk, keys):
+    dsk2 = cull(dsk, keys)
+    return fuse(dsk2)
+
+
+def get(dsk, keys, get=None, **kwargs):
+    """Specialized get function"""
+    get = get or _globals['get'] or threaded.get
+    dsk2 = optimize(dsk, keys)
+    return get(dsk2, keys, **kwargs)
+
+
+def right(method):
+    """Wrapper to create 'right' version of operator given left version"""
+    def _inner(self, other):
+        return method(other, self)
+    return _inner
+
+
+class Value(object):
+    """Represents a value to be computed by dask.
+
+    Equivalent to the output from a single key in a dask graph.
+    """
+    __slots__ = ('_key', '_dasks')
+
+    def __init__(self, name, dasks):
+        object.__setattr__(self, '_key', name)
+        object.__setattr__(self, '_dasks', dasks)
+
+    def compute(self, **kwargs):
+        """Compute the result."""
+        dask1 = merge(*self._dasks)
+        dask2 = cull(dask1, self.key)
+        return get(dask2, self.key, **kwargs)
+
+    @property
+    def dask(self):
+        return merge(*self._dasks)
+
+    @property
+    def key(self):
+        return self._key
+
+    def visualize(self, optimize_graph=False, **kwargs):
+        """Visualize the dask as a graph"""
+        from dask.dot import dot_graph
+        if optimize_graph:
+            dot_graph(optimize(self.dask, self.key), **kwargs)
+        else:
+            dot_graph(self.dask, **kwargs)
+
+    def __repr__(self):
+        return "Value({0})".format(repr(self._key))
+
+    def __hash__(self):
+        return hash(self.key)
+
+    def __dir__(self):
+        return list(self.__dict__.keys())
+
+    def __getattr__(self, attr):
+        if not attr.startswith('_'):
+            return do(getattr)(self, attr)
+        else:
+            raise AttributeError("Attribute {0} not found".format(attr))
+
+    def __setattr__(self, attr, val):
+        raise TypeError("Value objects are immutable")
+
+    def __iter__(self):
+        raise TypeError("Value objects are not iterable")
+
+    def __call__(self, *args, **kwargs):
+        return do(apply)(self, args, kwargs)
+
+    def __bool__(self):
+        raise TypeError("Truth of Value objects is not supported")
+
+    __nonzero__ = __bool__
+
+    __abs__ = do(operator.abs)
+    __add__ = do(operator.add)
+    __and__ = do(operator.and_)
+    __concat__ = do(operator.concat)
+    __delitem__ = do(operator.delitem)
+    __div__ = do(operator.floordiv)
```

Comment:
Why? `__div__` is only called on floordiv, `__truediv__` is called if future division was imported. Or is there some small difference I'm missing?